src/API/Google/AdsAssetGroup.php

```diff
@@ -190,11 +205,80 @@ protected function listing_group_delete_operations(): array {
 	}
 
 	/**
-	 * Return a temporary resource name for the campaign.
+	 * Return a temporary resource name for the asset group.
 	 *
 	 * @return string
 	 */
 	protected function temporary_resource_name() {
 		return ResourceNames::forAssetGroup( $this->options->get_ads_id(), self::TEMPORARY_ID );
 	}
+
+	/**
+	 * Get Asset Groups for a specific campaign.
+	 *
+	 * @since x.x.x
+	 *
+	 * @param int  $campaign_id The campaign ID.
+	 * @param bool $include_assets Whether to include the assets in the response.
+	 *
+	 * @return array The asset groups for the campaign.
+	 */
+	public function get_asset_groups_by_campaign_id( int $campaign_id, bool $include_assets = true ): array {
+		$asset_groups_converted = [];
+
+		$asset_group_results = ( new AdsAssetGroupQuery( [ 'pageSize' => self::PAGE_SIZE ] ) )
+			->set_client( $this->client, $this->options->get_ads_id() )
+			->add_columns( [ 'asset_group.path1', 'asset_group.path2', 'asset_group.id', 'asset_group.final_urls' ] )
+			->where( 'campaign.id', $campaign_id )
+			->where( 'asset_group.status', 'REMOVED', '!=' )
+			->get_results();
+
+		/** @var GoogleAdsRow $row */
+		foreach ( $asset_group_results->getPage()->getIterator() as $row ) {
+			$asset_groups_converted[ $row->getAssetGroup()->getId() ] = $this->convert_asset_group( $row );
```

Comment:
❓  Why we are saving the assets in `$asset_groups_converted` as an associative array if we return only the values ?

Reply:
I think it is handy to leave the `id` as a `key` because it allows retrieving the `ids` easily using `array_keys( $asset_groups );`.  For example in this case: https://github.com/woocommerce/google-listings-and-ads/blob/2ac0403841d865a5530546d8c87cdccbc364d8bf/src/API/Google/AdsAssetGroup.php#L258